Build fails with --enable-werror on some non-x86 architecture

When `--enable-werror ` is passed to configure script, the build fail on aarch64, armv6/7 and others.
The error log is about signed vs unsigned char:
```
[   78s] libqpdf/QPDFTokenizer.cc: In member function 'void QPDFTokenizer::findEI(PointerHolder<InputSource>)':
[   78s] libqpdf/QPDFTokenizer.cc:722:38: error: conversion to 'signed char' from 'char' may change the sign of the result [-Werror=sign-conversion]
[   78s]   722 |                     signed char ch = *iter;
[   78s]       |                                      ^~~~~
[   78s] libqpdf/QPDFTokenizer.cc:732:54: error: conversion to 'char' from 'signed char' may change the sign of the result [-Werror=sign-conversion]
[   78s]   732 |                     else if ((ch < 32) && (! isSpace(ch)))
[   78s]       |                                                      ^~
[   78s] cc1plus: all warnings being treated as errors
```

--- a/libqpdf/QPDFTokenizer.cc
+++ b/libqpdf/QPDFTokenizer.cc
@@ -719,7 +719,7 @@ QPDFTokenizer::findEI(PointerHolder<InputSource> input)
                 for (std::string::iterator iter = value.begin();
                      iter != value.end(); ++iter)
                 {
-                    signed char ch = *iter;
+                    char ch = *iter;
                     if (((ch >= 'a') && (ch <= 'z')) ||
                         ((ch >= 'A') && (ch <= 'Z')) ||
                         (ch == '*'))
@@ -729,10 +729,11 @@ QPDFTokenizer::findEI(PointerHolder<InputSource> input)
                         // alphabetic characters.
                         found_alpha = true;
                     }
-                    else if ((ch < 32) && (! isSpace(ch)))
+                    else if ((static_cast<signed char>(ch) < 32) &&
+                             (! isSpace(ch)))
                     {
-                        // ch is signed, so characters outside of
-                        // 7-bit will be < 0.
+                        // Compare ch as a signed char so characters
+                        // outside of 7-bit will be < 0.
                         found_non_printable = true;
                         break;
                     }

